### ISSUE TITLE
Shipping labels No rates available error for certain addresses

### ISSUE THREAD

**Issue by AashikP** [Automattic/woocommerce-services#2197]
[WooCommerce Services](https://woocommerce.com/products/shipping/?aff=10486&cid=1131038) is having trouble pulling rates for shipping labels for certain countries. Eg: Ghana, Nigeria.

From Address: (use any example US address)
Example to address: 28 Kodesoh St, Computer Village, Ikeja, Nigeria (100271)

Attempt to print a shipping label for the order and notice the first error:

![https://d.pr/i/rAiHfN+](https://d.pr/i/rAiHfN+)
*If the image does not load, please click this link: https://d.pr/i/rAiHfN*

When you finally add placeholder values:

![https://d.pr/i/yaybWn+](https://d.pr/i/yaybWn+)

---

While we initially thought that it is likely because USPS does not ship to those locations, user in 3276184-zen confirmed that they were able to mail both the packages from the post office.

Also, taking a look at https://about.usps.com/newsroom/service-alerts/international/welcome.htm

* Ghana should be available now: 
    ![https://d.pr/i/h5jjub+](https://d.pr/i/h5jjub+)
* And Nigeria isn't on the list at all.

---

Marking critical as depending on the address, the extension cannot be used at all for printing shipping labels.

**Comment by c-shultz** [comment]
@harriswong is going to follow up with EasyPost about this. They had told us that USPS cannot ship to those locations right now, but it sounds like that isn't the case based on that ticket.

**Comment by harriswong** [comment]
Reply from EasyPost:
> ...EasyPost has actually disabled USPS rates to Nigeria and Ghana due to the amount of fraud with these countries in the past. 

@c-shultz We can display some notification after address check, saying that we don't support shipping to these countries for USPS, you will need to visit a postal office physically. At the same time, i also think that we can close this issue without making changes because we are showing no USPS rates at the moment. "No rates available" is the same as "no shipping available to these countries".

**Comment by c-shultz** [comment]
I agree with closing this. Let's discuss best next steps on P2. At a minimum, we should have a note in our docs.

**Comment by harriswong** [comment]
We discussed in team and we are going to re-open this. 

While we can't fetch rates for these 2 countries, we would also like to prevent confusion to the users. Instead of just not displaying any rates. We agree to display a message when the destination is being shipped to these 2 countries. Something similar to this:
![image](https://user-images.githubusercontent.com/572862/95639478-cb0c2200-0a55-11eb-82f1-9c14702aa7ce.png)

- We should not block other carriers from displaying rates. For example, DHL rates should still show, but with this error message on top specifically for USPS.
- We don't need the "learn more" link if EasyPost does not have a pubic documentation page for it.
- If we can't use the "learn more" link, then the copy would need to provide some more context. It will have to be something more specific on why and/or what options there are.

**Comment by c-shultz** [comment]
I'm going to change this priority to high. This is expected behavior based on what we've learned from EasyPost.

**Comment by harriswong** [comment]
We chatted in our team meeting. When fixing this, we should consider:
- The logic that decides "when to show the banner message" should be on the server side. ie. if country==='ghana', then server will return [] rates with a banner message.
- If EasyPost lifted the restriction, we should show the rate without us changing the code. Thus, checking against just country code is not enough.

**Comment by efixler** [comment]
I've dropped priority on this issue. 

EasyPost will determine lanes for which fraud is likely. TBH, we should we doing the same; account takeover + stolen cards used to purchase international labels to specific countries is a known problem in the space.

The most important behavior here is that we don't provide labels to countries tagged as fraud-likely or forbidden by our upstream providers or carriers. This is already the case.

I'm willing to speculate that the number of legitimate requests for US->Nigeria/Ghana/etc rates is infinitesimal. As such, I don't see this as a high priority issue, at least when there are other issues in the backlog where there are known cases of providing incorrect rates.

**Comment by Abdalsalaam** [comment]
Shipping related work now happens in WooCommerce Shipping and we recommend using it: https://wordpress.org/plugins/woocommerce-shipping/
